Currently this is failing CI type checks with the error: `Cannot find module '@automattic/calypso-url' or its corresponding type declarations.`. That doesn't make sense since that package is listed correctly as a dependency and there are no type errors locally.

My guess is that this is because alphabetically, this package (`calypso-products`) comes before the dependency (`calypso-url`), so maybe the dependency is not yet built and is therefore giving an error.

I'm asking in Slack to see if there's any ideas for resolving this issue.

--- a/packages/calypso-products/package.json
+++ b/packages/calypso-products/package.json
@@ -38,7 +38,6 @@
 	"homepage": "https://github.com/Automattic/wp-calypso/tree/HEAD/packages/calypso-products#readme",
 	"dependencies": {
 		"@automattic/calypso-config": "workspace:^",
-		"@automattic/calypso-url": "workspace:^",
 		"i18n-calypso": "workspace:^",
 		"react": "^17.0.2"
 	},

--- a/packages/calypso-products/src/main.ts
+++ b/packages/calypso-products/src/main.ts
@@ -1,11 +1,4 @@
 import {
-	format as formatUrl,
-	getUrlParts,
-	getUrlFromParts,
-	determineUrlType,
-	URL_TYPE,
-} from '@automattic/calypso-url';
-import {
 	TERM_MONTHLY,
 	TERM_ANNUALLY,
 	TERM_BIENNIALLY,
@@ -441,35 +434,22 @@ export function getBillingMonthsForTerm( term: string ): number {
 }
 
 export function plansLink(
-	url: string,
+	urlString: string,
 	siteSlug: string,
 	intervalType: string,
 	forceIntervalType = false
 ): string {
-	const originalUrlType = determineUrlType( url );
-	const resultUrl = getUrlParts( url );
-
-	if ( originalUrlType === URL_TYPE.INVALID ) {
-		throw new Error( 'Cannot format an invalid URL.' );
-	}
-
-	if ( originalUrlType === URL_TYPE.PATH_RELATIVE ) {
-		throw new Error( 'Cannot format path-relative URLs.' );
-	}
+	const url = new URL( urlString, window.location.origin );
 
 	if ( 'monthly' === intervalType || forceIntervalType ) {
-		resultUrl.pathname += '/' + intervalType;
+		url.pathname += '/' + intervalType;
 	}
 
 	if ( siteSlug ) {
-		resultUrl.pathname += '/' + siteSlug;
+		url.pathname += '/' + siteSlug;
 	}
 
-	// getUrlFromParts only works with absolute URLs, so add some dummy data if
-	// needed, and format down to the type of the original url.
-	resultUrl.protocol = resultUrl.protocol || 'https:';
-	resultUrl.hostname = resultUrl.hostname || '__hostname__.invalid';
-	return formatUrl( getUrlFromParts( resultUrl ), originalUrlType );
+	return url.toString();
 }
 
 export function applyTestFiltersToPlansList(

--- a/packages/calypso-products/src/main.ts
+++ b/packages/calypso-products/src/main.ts
@@ -449,6 +449,9 @@ export function plansLink(
 		url.pathname += '/' + siteSlug;
 	}
 
+	if ( urlString.startsWith( '/' ) ) {
+		return url.pathname + url.search;
+	}
 	return url.toString();
 }
 

--- a/packages/calypso-products/test/plans-link.js
+++ b/packages/calypso-products/test/plans-link.js
@@ -1,8 +1,12 @@
 import { plansLink } from '../src';
 
 describe( 'plansLink', () => {
-	test( 'should return url unchanged for yearly no-site', () => {
+	test( 'should return relative url unchanged for yearly no-site', () => {
 		expect( plansLink( '/plans' ) ).toBe( '/plans' );
+	} );
+
+	test( 'should return absolute url unchanged for yearly no-site', () => {
+		expect( plansLink( 'https://example.com/plans' ) ).toBe( 'https://example.com/plans' );
 	} );
 
 	test( 'should append monthly to url when required', () => {

